### PR TITLE
[FCE-686]: Allow to reconfigure ForegroundService message

### PR DESCRIPTION
**Pull request by MiloszFilimowski**
## Description

- App won't crash/emit warning when `toggleScreenShare` is used without `useForegroundService` (default are now used to schedule the foreground service).
- Notification content gets updated when params change

## Motivation and Context

- It was not possible to update notification when already created.

## How has this been tested?

- Tested on emulator API 35/33
- Tested on real device API 34

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/react-native-client/src/hooks/useForegroundService.ts [hunk 1]
```diff
@@ -65,13 +55,24 @@ const useForegroundServiceAndroid = ({
...
+    enableMicrophone,
+    isConfigured,
+    notificationContent,
+    notificationTitle,
+  ]);
 
   useEffect(() => {
     const runConfiguration = async () => {
       await requestNotificationsPermission();
-      RNFishjamClientModule.configureForegroundService(restOptions);
```
Comment (mironiasty) [inline packages/react-native-client/src/hooks/useForegroundService.ts:74]: `eslint-disable-next-line` comment below is no longer requierd

File: packages/react-native-client/src/hooks/useScreenShare.ts [hunk 2]
```diff
@@ -94,10 +94,9 @@ export function useScreenShare() {
   const toggleScreenShare = useCallback(
     async (screenShareOptions: Partial<ScreenShareOptions> = {}) => {
       if (Platform.OS === 'android' && !isScreenShareOn) {
-        (await handleScreenSharePermission()) == 'granted' &&
-          (await RNFishjamClientModule.startForegroundService({
-            enableScreenSharing: true,
-          }));
+        if ((await handleScreenSharePermission()) != 'granted') {
```
Comment (mironiasty) [inline packages/react-native-client/src/hooks/useScreenShare.ts:97]: WDYT about enabling eslint rule that forces `===` and `!==` 🤔

File: packages/react-native-client/android/src/main/java/io/fishjam/reactnative/foregroundService/ForegroundServiceState.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      if (screenSharingEnabled) add(FOREGROUND_SERVICE_TYPE_MEDIA_PROJECTION)
+    }
+
+  private fun MutableList<Int>.addIfPermissionGranted(
+    serviceType: Int,
+    permissionName: String,
+    hasPermission: () -> Boolean
+  ) {
+    if (!hasPermission()) {
+      throw CodedException("Cannot start a $permissionName foreground service without $permissionName permission.")
```
Comment (mironiasty) [inline packages/react-native-client/android/src/main/java/io/fishjam/reactnative/foregroundService/ForegroundServiceState.kt:42]: What will happen with app, when this exception is thrown?

File: packages/react-native-client/src/hooks/useScreenShare.ts (hunk 2 above)
Comment (MiloszFilimowski) [reply packages/react-native-client/src/hooks/useScreenShare.ts:97]: Good point! It's easy to miss this when switching between native/js side.

File: packages/react-native-client/android/src/main/java/io/fishjam/reactnative/foregroundService/ForegroundServiceState.kt (hunk 3 above)
Comment (MiloszFilimowski) [reply packages/react-native-client/android/src/main/java/io/fishjam/reactnative/foregroundService/ForegroundServiceState.kt:42]: Promise will be rejected with this error.

File: packages/react-native-client/android/src/main/java/io/fishjam/reactnative/foregroundService/ForegroundServiceManager.kt [hunk 4]
```diff
@@ -4,79 +4,77 @@ import android.content.ComponentName
...
+    )
 
   private val connection =
     object : ServiceConnection {
       override fun onServiceConnected(
         className: ComponentName,
         service: IBinder
       ) {
-        isServiceBound = true
+        serviceInstance = (service as FishjamForegroundService.LocalBinder).getService()
```
[the remark this answers is not in the document]
Comment (MiloszFilimowski) [reply packages/react-native-client/android/src/main/java/io/fishjam/reactnative/foregroundService/ForegroundServiceManager.kt:38]: This is an override, so I can't change the function signature.

**Review by maksg — APPROVED** [review]

**Pull request merged**